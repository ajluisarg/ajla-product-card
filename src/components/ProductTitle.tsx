import React, { useContext } from "react";
import { ProductContext } from "./ProductCard";
import styles from "../styles/styles.module.css";

export const ProductTitle = ({title: providedTitle, className}: {title?: string, className?: string}) => {
    const {product: {title} = {title: ''}} = useContext(ProductContext);
  
    return <span className={`${styles.productDescription} ${className}`}>{providedTitle || title}</span>;
  };