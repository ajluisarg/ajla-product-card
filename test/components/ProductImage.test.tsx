import React from 'react';
import renderer from 'react-test-renderer';
import { ProductCard, ProductImage } from '../../src/components';
import { product2 } from './data/products';

describe('ProductImage', () => {

    test('show component', () => {

        const wrapper = renderer.create(
            <ProductImage img="test2.png"/>
        )

        expect(wrapper.toJSON()).toMatchSnapshot();
        
    });

    test('show product img from context', () => {
        const wrapper = renderer.create(
            <ProductCard product={product2}>
                {
                    () => (
                        <ProductImage/>
                    )
                }
            </ProductCard>
        )

        expect(wrapper.toJSON()).toMatchSnapshot();
    })
});
