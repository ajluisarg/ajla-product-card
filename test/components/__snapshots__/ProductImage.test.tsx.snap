// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage show component 1`] = `
<img
  alt="image"
  className="productImg undefined"
  src="test2.png"
/>
`;

exports[`ProductImage show product img from context 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Coffee"
    className="productImg undefined"
    src="/public/test.png"
  />
</div>
`;
